dind/Dockerfile

```diff
@@ -1,26 +1,35 @@
 ARG DOCKER_VERSION
 
-FROM docker:${DOCKER_VERSION}-dind-rootless
+FROM golang:alpine AS build
```

Comment:
We should fix the golang image to the version specified in `go.mod`, maybe is a good idea to have an ARG as the one for DOCKER_VERSION.

Reply:
Not sure if it's a good idea to add the ARG, as the version would still be fixed in `go.mod`.

Reply:
a2643b2

Reply:
Indeed, we can keep it hardcoded then